frontend/src/components/Chat/ChatSidebar.tsx

```diff
@@ -1,48 +1,181 @@
-import { Tab, TabList, TabPanel, TabPanels, Tabs } from '@chakra-ui/react';
-import React, { useState } from 'react';
+import { Button, Tab, TabList, TabPanel, TabPanels, Tabs, Tag } from '@chakra-ui/react';
+import React, { useCallback, useMemo, useState } from 'react';
 import { MessageType } from '../../classes/MessageChain';
+import useCoveyAppState from '../../hooks/useCoveyAppState';
 import ChatContainer from './ChatContainer';
 import './ChatSidebar.css';
 import DirectMessageSelect from './DirectMessageSelect';
 
 export default function ChatSidebar(): JSX.Element {
   const [isViewingDirectChatContainer, setIsViewingDirectChatContainer] = useState<boolean>(false);
+  const [isCollapsed, setIsCollapsed] = useState<boolean>(true);
+  const [chatName, setChatName] = useState<string>('Direct Chats');
+  const { townMessageChain, proximityMessageChain, directMessageChains } = useCoveyAppState();
+  const numberUnviewedTownMessages = townMessageChain.numberUnviewed;
+  const numberUnviewedProximityMessages = proximityMessageChain.numberUnviewed;
+  const numberUnviewedDirectMessages = Object.values(directMessageChains).reduce(
+    (acc, currentChain) => acc + currentChain.numberUnviewed,
+    0,
+  );
+
+  const onTownChatSelect = useCallback(() => setChatName('Town Chat'), [setChatName]);
+  const onProximityChatSelect = useCallback(() => setChatName('Proximity Chat'), [setChatName]);
+  const onDirectChatSelect = useCallback(() => setChatName('Direct Chats'), [setChatName]);
+  const onDirectChatOpen = useCallback(userName => setChatName(`Direct Chat with ${userName}`), [
+    setChatName,
+  ]);
+
+  const townMessageNotifs = useMemo(() => {

```

Comment:
`useMemo` will prevent re-renders of these elements so long as the dependencies (the stuff in the array below) remain unchanged. this is a critical performance improvement in apps where data updates often